### DISCUSSION TITLE
grpc.FutureTimeoutError when connecting to milvus docker

### DISCUSSION THREAD

**Discussion by Evennaire** [milvus-io/milvus#32260] [Q&A and General discussion] [answered]
when trying to connect to milvus this error occurs:
```
Traceback (most recent call last):
  File "/opt/anaconda3/envs/py39/lib/python3.9/site-packages/pymilvus/client/grpc_handler.py", line 143, in _wait_for_channel_ready
    grpc.channel_ready_future(self._channel).result(timeout=timeout)
  File "/opt/anaconda3/envs/py39/lib/python3.9/site-packages/grpc/_utilities.py", line 162, in result
    self._block(timeout)
  File "/opt/anaconda3/envs/py39/lib/python3.9/site-packages/grpc/_utilities.py", line 106, in _block
    raise grpc.FutureTimeoutError()
grpc.FutureTimeoutError
```

Checked the logs with several errors:
```
milvus-standalone  | [2024/04/15 06:39:27.725 +00:00] [WARN] [grpcclient/client.go:476] ["fail to get grpc client in the retry state"] [client_role=rootcoord] [error="failed to connect 172.19.0.4:53100, reason: context deadline exceeded"]
milvus-standalone  | [2024/04/15 06:39:30.081 +00:00] [ERROR] [sessionutil/session_util.go:475] ["retry func failed"] ["retry time"=4] [error="function CompareAndSwap error for compare is false for key: datacoord"]
milvus-standalone  | [2024/04/15 06:39:30.932 +00:00] [WARN] [retry/retry.go:44] ["grpc client is nil, maybe fail to get client in the retry state"] [client_role=rootcoord] [error="empty grpc client: failed to connect 172.19.0.4:53100, reason: context deadline exceeded"]
```

Here is the full log file: 
[milvus.log](https://github.com/milvus-io/milvus/files/14975797/milvus.log)

**Comment by yhmo** [comment]
The log is from 2024/04/15 06:39:26 to 2024/04/15 06:40:00. Seems the milvus is still in initialize, no critical errors.
The python sdk connect() method timeout value by default is 10 seconds, so you got the FutureTimeoutError. You can try connect again after a few seconds.

If you see "---Milvus Proxy successfully initialized and ready to serve!---" in the log, that means the milvus is ready to work.

**Reply by Evennaire** [reply]
I updated the log file. The grpc.FutureTimeoutError still exists after initiated.

**Reply by yhmo** [reply]
This is a bug of 2.3.11, if a milvus restarts, the previous information in etcd could affect the later. Restart it again can solve this problem.
This bug is fixed in v2.3.13, you can upgrade to 2.3.13 to avoid this bug.

**Reply by Evennaire** [reply]
I have deleted the volumes, upgraded to 2.3.13 and restarted milvus docker. The error still exists, with new log:
[milvus.log](https://github.com/milvus-io/milvus/files/14981302/milvus.log)
Should I do other things to clean the previous information? Pls help me

**Reply by xiaofan-luan** [reply, marked as answer]
1. what milvus server version and pymilvus version are you using?
2. Form the log, server side is working as expected. Could you ping to the endpoint see if network is accessessible?

**Reply by yhmo** [reply]
The new log seems no problem.

**Reply by Evennaire** [reply]
Solved, it was the network problem. The port was not open properly. Thank you!

**Comment by christy** [comment]
@Evennaire - there's probably more to your bug, but I noticed the [milvus docker page](https://milvus.io/docs/configure-docker.md.) was updated just 2 weeks ago, it was buggy before that.  

For the source of truth, download `docker-compose.yml` directly from Milvus github.  I just tried again and it worked.  Checked in [example code](https://github.com/milvus-io/bootcamp/blob/master/bootcamp/milvus_connect.ipynb) on github.  (If you use it, change my wget command to 2.3.13 if that's what you're using).

Remember Pymilvs and server versions have to match Major.Minor versions.  Such as Pymilvus 2.3.6 with Milvus local docker server 2.3.13.  Even though the latest Pymilvus is v2.4, we don't recommend it yet for production.

**Reply by Evennaire** [reply]
It was my network problem. But I have updated my yml file based on new docker page and your example code, and now it works good. Thank you!
